src/Chapter1.hs

```diff
@@ -542,7 +545,10 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid x y z 
+    | x < y = if y < z then y else z
+    | z < x = if y < z then z else y
+    | otherwise = z 
```

Comment:
I noticed, that this function never returns `x`. Probably, needs to modification, as you may want to have something like:

```haskell
ghci> mid 2 1 3
2
```